internal/execute/watcher.go

```diff
@@ -73,12 +75,14 @@ func (w *watcher) hasBeenModified(program *compiler.Program) bool {
 		currState[fileName] = s.ModTime()
 		if !filesModified {
 			if currState[fileName] != w.prevModified[fileName] {
+				// fmt.Fprint(w.sys.Writer(), "build triggered from ", fileName, ": ", w.prevModified[fileName], " -> ", currState[fileName], w.sys.NewLine())
 				filesModified = true
 			}
-			delete(w.prevModified, fileName)
 		}
+		delete(w.prevModified, fileName)
```

Comment:
It was intentional to leave this inside the `if !filesModified` block, since the deletion only matters if `filesModified == false`. You could add a comment here with this reasoning